Dockerfile

```diff
@@ -251,24 +251,24 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
 
 FROM base AS dockercli
 WORKDIR /go/src/github.com/docker/cli
-COPY hack/dockerfile/cli.sh /download-or-build-cli.sh
 ARG DOCKERCLI_REPOSITORY
 ARG DOCKERCLI_VERSION
 ARG TARGETPLATFORM
-RUN --mount=type=cache,id=dockercli-git-$TARGETPLATFORM,sharing=locked,target=./.git \
+RUN --mount=source=hack/dockerfile/cli.sh,target=/download-or-build-cli.sh \
```

Comment:
Does the build-cache take changes in mounted files into account? If not, then I'm not sure if this is the right thing to do, because it would then silently ignore change in the `cli.sh` script (which could be, e.g. "use a different repo").

Reply:
Ah, perfect; I wasn't 100% sure (I guess only "cache" mounts are ignored) 👍

Reply:
Also the thing to keep in mind with `RUN --mount=type=bind` is that it's a bind mount from the context, not from the filesystem; so a change in the build context will bust all dependents (`COPY` and `--mount`).